src/Microsoft.AspNetCore.Razor.Tasks/DotnetToolTask.cs

```diff
@@ -82,10 +103,102 @@ protected override void LogToolCommand(string message)
             }
         }
 
-        protected virtual bool TryExecuteOnServer(out bool result)
+        public override void Cancel()
         {
-            result = false;
+            base.Cancel();
+
+            _razorServerCts?.Cancel();
+        }
+
+        protected virtual bool TryExecuteOnServer(string pathToTool, string responseFileCommands, string commandLineCommands, out int result)
+        {
+            CompilerServerLogger.Log("Server execution started.");
+            using (_razorServerCts = new CancellationTokenSource())
+            {
+                CompilerServerLogger.Log($"CommandLine = '{commandLineCommands}'");
+                CompilerServerLogger.Log($"BuildResponseFile = '{responseFileCommands}'");
+
+                var clientDir = Path.GetDirectoryName(ServerAssembly);
```

Comment:
Add a comment about this expectation that the client bits exist in the server bits (that we ship both together)